src/main/java/org/mule/tools/apikit/model/CustomConfiguration.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.tools.apikit.model;
+
+import java.util.List;
+import java.util.Optional;
+
+public class CustomConfiguration {
+
+  private Optional<String> externalConfigurationFile;
+  private Optional<List<String>> environments;
+  private Optional<String> apiAutodiscoveryID;
+
+  public CustomConfiguration() {
+    this.externalConfigurationFile = Optional.empty();
+    this.environments = Optional.empty();
+    this.apiAutodiscoveryID = Optional.empty();
+  }
+
+  public CustomConfiguration(Optional<String> externalConfigurationFile, Optional<List<String>> environments,
```

Comment:
Just a kind reminder, avoid using Optional as arguments. 
Here is a list of (some of the) good reasons
https://stackoverflow.com/questions/31922866/why-should-java-8s-optional-not-be-used-in-arguments

Reply:
I made proper changes, it receives String and makes it optional inside.